### ISSUE TITLE
LinuxVirtualMachineScaleSet should allow UpgradePolicy=Automatic without requiring health_probe_id

### ISSUE THREAD

**Issue by kuba-lilz** [pulumi/pulumi-azure#682]
language: python 3.8
pulumi version: v2.9.0
pulumi.azure version: 3.17.0

TLDR: adding scale set to application gateway's backend pool requires upgrade to scale set, but to use automatic upgrades we are asked to provide load balancer's health check - even though scale set is connected to application gateway, not load balancer.

To set `upgrade_mode` of `LinuxVirtualMachineScaleSet` to `Automatic`, one has to provide a bunch of other options, including `health_probe_id` that is supposed to point to a load balancer's health probe.

The problem is that there are valid actions requiring upgrades to `LinuxVirtualMachineScaleSet` that have nothing to do with load balancers.
An example I run into is attaching `LinuxVirtualMachineScaleSet` to `ApplicationGateway` (through settings scaling set's network interface to contain desired backend pool id in its `applicationGatewayBackendAddressPoolIds`)

Setting this property after resource was created (modifying resource), causes scale set model and virtual machine model to diverge on network configuration. For virtual machines to match scale set model, `Upgrade` action has to be triggered.
This is a simple upgrade that doesn't reboot machines. I would like to be able to set this upgrade to Automatic mode, but without a load balancer, which I don't need in this scenario, Pulumi won't let me set upgrades to automatic mode.
And until I upgrade virtual machines, application gateway won't accept them into the backend pool.
Then end result is that while Pulumi offers way to attach scaling sets to application gateway backend pools, they end up in `awaiting upgrade` state rather than really attached.

Btw, terraform allows setting upgrade policy to automatic without defining any health probes.

**Comment by kuba-lilz** [comment]
At very it would be great to be able to use application gateway's health probe, but that's not supported - only load balancer's health probe is accepted.

**Comment by kuba-lilz** [comment]
@viveklak 
Is there any timeline for when this issue might be resolved? This is stopping us from switching from Terraform to Pulumi in production.

**Comment by viveklak** [comment]
@kuba-lilz Apologies for the delay. I plan on looking into this this week. Would you be able to share a code snippet demonstrating the issue?

**Comment by kuba-lilz** [comment]
So I tried putting together a minimal example with latest pulumi version and... turns out my problem is already solved 😀 

Just for reference, here's the minimal example that tries to use "upgrade_mode": "Automatic" policy on a scaling set:

```python
ip_configuration = {
            "name": "...",
            "primary": True,
            "subnet_id": "..."
        }

network_interface = {
    "name": "...",
    "primary": True,
    "ip_configurations": [ip_configuration],
    "network_security_group_id": "..."
}

scale_set = pulumi_azure.compute.LinuxVirtualMachineScaleSet(
    resource_name="example",
    resource_group_name=...,
    sku="Standard_F2s_v2",
    instances=2,
    admin_username="...",
    admin_ssh_keys=[{
        "username": "...",
        "publicKey": "...",
    }],
    source_image_id=...
    os_disk={...},
    network_interfaces=[network_interface],
    upgrade_mode="Automatic"
)
```

With python packages: `pulumi==2.9.0 pulumi-azure==3.17.0`
above code throws an error during `pulumi up` with message:
```
  azure:compute:LinuxVirtualMachineScaleSet (example):
    error: `healthProbeId` must be set when `upgrade_mode` is set to "Automatic"
```

But upgrading to `pulumi==2.9.2 pulumi-azure==3.20.0`, code works as expected and scaling set is created with automatic upgrade policy.
I don't know which change between pulumi 2.9.0 -> 2.9.2 and pulumi-azure 3.17.0 -> 3.20.0 relaxed `healthProbeId` requirement that was blocking us, but hey, we can move forward with Pulumi in production now!
